queue/worker.go

```diff
@@ -1,57 +1,84 @@
 package queue
 
 import (
+	"fmt"
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
+
+	"github.com/goravel/framework/contracts/database/orm"
 	"github.com/goravel/framework/contracts/queue"
+	"github.com/goravel/framework/support/carbon"
 )
 
-const DriverSync string = "sync"
-const DriverRedis string = "redis"
-
 type Worker struct {
 	concurrent int
-	connection string
-	machinery  *Machinery
-	jobs       []queue.Job
+	driver     queue.Driver
+	failedJobs orm.Query
 	queue      string
 }
 
-func NewWorker(config *Config, concurrent int, connection string, jobs []queue.Job, queue string) *Worker {
+func NewWorker(config *Config, concurrent int, connection string, queue string) *Worker {
 	return &Worker{
 		concurrent: concurrent,
-		connection: connection,
-		machinery:  NewMachinery(config),
-		jobs:       jobs,
+		driver:     NewDriver(connection, config),
+		failedJobs: config.FailedJobsDatabase(),
 		queue:      queue,
 	}
 }
 
-func (receiver *Worker) Run() error {
-	server, err := receiver.machinery.Server(receiver.connection, receiver.queue)
-	if err != nil {
-		return err
-	}
-	if server == nil {
-		return nil
+func (r *Worker) Run() error {
+	if r.driver.DriverName() == DriverSync {
+		return fmt.Errorf("queue %s driver not need run", r.queue)
 	}
 
-	jobTasks, err := jobs2Tasks(receiver.jobs)
-	if err != nil {
-		return err
-	}
+	failedJobChan := make(chan FailedJob)
+	sigChan := make(chan os.Signal, 1)
+	quitChan := make(chan struct{})
+	var wg sync.WaitGroup
+	signal.Notify(sigChan, syscall.SIGHUP, syscall.SIGINT, syscall.SIGQUIT, syscall.SIGTERM)
 
-	if err := server.RegisterTasks(jobTasks); err != nil {
-		return err
+	for i := 0; i < r.concurrent; i++ {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			for {
+				select {
+				case <-quitChan:
+					return
+				default:
+					job, args, err := r.driver.Pop(r.queue)
+					if err != nil {
```

Comment:
S: There are a few different errors, we'd better handle it.

<img width="781" alt="image" src="https://github.com/goravel/framework/assets/24771476/17d05619-7935-493a-bcc8-8aad21777bc6">